lesson25/src/js/main.js

```diff
@@ -0,0 +1,54 @@
+import { toggleModal } from "./modal";
+import { validateForm } from "./validation";
+
+const agreeCheckBox = document.querySelector('[data-id="agree_check"]');
+const agreeButton   = document.querySelector('[data-id="agree_button"]');
+const submitButton  = document.querySelector('[data-id="submit_button"]');
+
+agreeButton.addEventListener('click' , () => {
+    agreeCheckBox.removeAttribute('disabled');
+    agreeCheckBox.classList.remove("field-invalid");
+    agreeCheckBox.checked = true;
+    changeDisabledStatusSubmitButton();
+});
+
+submitButton.addEventListener("click", (e) => {
+    e.preventDefault();
+    window.location.href = "register-done.html";
+});
+
+const options = {
+    root: document.querySelector('[data-id="modal-inner"]'),
+    threshold: 1
+};
+
+const removeDisabledForAgreeButton = ([entry]) => {
+    if(entry.isIntersecting){
+        const agreeButton  = document.querySelector('[data-id="agree_button"]');
+        agreeButton.removeAttribute('disabled');
+    }
+};
+
+const observer = new IntersectionObserver(removeDisabledForAgreeButton, options);
+observer.observe(document.querySelector('[data-id="last_text"]'));
+
+const invalidSelector = document.getElementsByClassName("field-invalid");
+const changeDisabledStatusSubmitButton = () => submitButton.disabled = !(invalidSelector.length === 0);
```

Comment:
nits
You can also write
```suggestion
const changeDisabledStatusSubmitButton = () => submitButton.disabled = invalidSelector.length > 0;
```

Reply:
Thank you for your advice.
I fixed!
https://github.com/nakagawamai/js-lesson/pull/27/commits/0b5709e3b37e47d8a10410a8421ff0466eb5cb79